Sources/PostgresNIO/New/PSQLConnection.swift

```diff
@@ -188,76 +198,79 @@ final class PSQLConnection {
         on eventLoop: EventLoop
     ) -> EventLoopFuture<PSQLConnection> {
         
-        let connectionID = "1"
+        let connectionID = UUID().uuidString
         var logger = logger
         logger[postgresMetadataKey: .connectionID] = "\(connectionID)"
         
-        return eventLoop.makeSucceededFuture(Void()).flatMapThrowing { _ -> SocketAddress in
-            switch configuration.connection {
-            case .resolved(let address, _):
-                return address
-            case .unresolved(let host, let port):
-                return try SocketAddress.makeAddressResolvingHost(host, port: port)
-            }
-        }.flatMap { address in
-            let bootstrap = ClientBootstrap(group: eventLoop)
-                .channelInitializer { channel in
-                    let decoder = ByteToMessageHandler(PSQLBackendMessage.Decoder())
-                    
-                    var enableSSLCallback: ((Channel) -> EventLoopFuture<Void>)? = nil
-                    if let tlsConfiguration = configuration.tlsConfiguration {
-                        enableSSLCallback = { channel in
-                            channel.eventLoop.submit {
-                                let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
-                                return try NIOSSLClientHandler(
-                                    context: sslContext,
-                                    serverHostname: configuration.sslServerHostname)
-                            }.flatMap { sslHandler in
-                                channel.pipeline.addHandler(sslHandler, position: .before(decoder))
+        return eventLoop.flatSubmit {
+            eventLoop.makeSucceededFuture(Void()).flatMapThrowing { _ -> SocketAddress in
+                switch configuration.connection {
+                case .resolved(let address, _):
+                    return address
+                case .unresolved(let host, let port):
+                    return try SocketAddress.makeAddressResolvingHost(host, port: port)
+                }
+            }.flatMap { address -> EventLoopFuture<Channel> in
```

Comment:
```suggestion
        return eventLoop.submit { () -> SocketAddress in
            switch configuration.connection {
            case .resolved(let address, _):
                return address
            case .unresolved(let host, let port):
                return try SocketAddress.makeAddressResolvingHost(host, port: port)
            }
            .flatMap { address -> EventLoopFuture<Channel> in
```
Adjust indentation appropriately throughout and remove the extra closing brace below.

Reply:
This is not semantically the same. With your proposed change all `flatMap`s are enqueued on the main thread. Whereas with my implementation all that enqueuing happens on the EL.

Reply:
Replaced: